File: leetcode/problems/24.ts
function swapPairs(head: ListNode | null): ListNode | null {
  head = new ListNode(0, head)
  let p = head.next,
    q = head
  while (p && p.next) {
    q.next = p.next
    const tmp = p.next.next
    p.next.next = p
    p.next = tmp
    q = p
    p = tmp
  }
  return head.next
}

File: leetcode/problems/offer/24.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func reverseList(head *ListNode) *ListNode {
  var pre *ListNode
  for head != nil {
    tmp := head.Next
    head.Next = pre
    pre = head
    head = tmp
  }
  return pre
}